internal/controller/postgrescluster/cluster_test.go

```diff
@@ -701,3 +702,59 @@ func TestContainerSecurityContext(t *testing.T) {
 		}
 	}
 }
+
+func TestGenerateClusterReplicaServiceIntent(t *testing.T) {
+	cluster := &v1beta1.PostgresCluster{}
+	cluster.Namespace = "ns1"
+	cluster.Name = "pg2"
+	cluster.Spec.Port = initialize.Int32(9876)
+
+	service := generateClusterReplicaServiceIntent(cluster)
+
+	assert.Assert(t, marshalMatches(service.TypeMeta, `
+apiVersion: v1
+kind: Service
+	`))
+	assert.Assert(t, marshalMatches(service.ObjectMeta, `
+creationTimestamp: null
+labels:
+  postgres-operator.crunchydata.com/cluster: pg2
+  postgres-operator.crunchydata.com/role: replica
+name: pg2-replicas
+namespace: ns1
+	`))
+	assert.Assert(t, marshalMatches(service.Spec, `
+ports:
+- name: postgres
+  port: 9876
+  protocol: TCP
+  targetPort: postgres
+selector:
+  postgres-operator.crunchydata.com/cluster: pg2
+  postgres-operator.crunchydata.com/role: replica
+type: ClusterIP
+	`))
+
+	t.Run("Labels", func(t *testing.T) {
```

Comment:
:pick: add in a test for cluster annotations. Looks like everything else is covered by the spec asserts